### PR TITLE
SCR:2497: Added err checks

### PR DESCRIPTION
**Pull request by anirudhmundra**

### CODE REVIEW

File: xf-goredis/cache.go [hunk 1]
```diff
@@ -22,25 +22,37 @@ func Wrap(client redis.UniversalClient) xf.Cache {
...
-		return nil, 0, 0, errors.Wrap(err, "error on executing pipeline")
+		if err != redis.Nil {
+			return nil, 0, 0, errors.Wrap(err, "error on executing pipeline")
+		}
 	}
 
 	value, err := readPipe.Result()
 	if err != nil {
-		return nil, 0, 0, errors.Wrap(err, "reading")
+		if err != redis.Nil {
```
Comment (jpatel531) [inline xf-goredis/cache.go:37]: maybe you could try something like this? https://github.com/Onefootball/xfetch-go/blob/master/xf-redigo/cache.go#L56

Comment (jpatel531) [reply xf-goredis/cache.go:37]: the early return is more readable

**Review by jpatel531 — CHANGES_REQUESTED** [review]

File: xf-goredis/cache.go (hunk 1 above)
Comment (anirudhmundra) [reply xf-goredis/cache.go:37]: @jpatel531 
i went with an approach to become closer to redigo implementation
and for the `value` having nil we do not return 
we continue to parse ttl and delta

but if we can return without checking for ttl & delta - yes we could change it

Comment (jpatel531) [reply xf-goredis/cache.go:37]: yea i think it's a bit strange to have it done one way (on line 49) then for it to be done differently here. also not sure about doing `value = nil`

**Review by jpatel531 — APPROVED** [review]

**Pull request merged**